app/models/person.rb

```diff
@@ -113,6 +113,11 @@ class Person < ActiveRecord::Base
 
   has_and_belongs_to_many :followed_listings, :class_name => "Listing", :join_table => "listing_followers"
 
+  after_save do
+    # This is required to invalidate the fragment caches for listings on front page, since the author is shown there.
+    self.listings.all.map(&:touch)
+  end
+
```

Comment:
This is indeed quite problematic. I'd suggest looking for these two options:

1. Can we do this in Controller layer? The model hooks are quite problematic, and some people even consider them as anti-pattern, so adding manual invalidation in the controller would solve that.
2. Listing.where(author_id: <author_id).update_all(updated_at: Time.now) can do the update in one query.